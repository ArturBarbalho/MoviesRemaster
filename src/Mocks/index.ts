import { BannerProps, CardProps, bigbannerProps } from "../typesProps";
export const BiggerBannerMock:bigbannerProps = {
  bigsrc: 'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
  smallsrc:'https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg',
  genres:['Action', 'Comedy'],
  id:2,
  media:'movie',
  overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
  title:'ONYANKOPON 3 O IN√çCIO'
}
export const CardMock:CardProps = {
  id:2,
  media:'movie',
  src:'http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg',
  title:'ONYANKOPON 2 O RETORNO'
}

export const BannerMock:BannerProps = {
    media: 'movie',
    relaseDate: '(1800)',
    id:1,
    title:'ONYANKOPON',
    src:'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
    overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
    genres: ['Action', 'Comedy' ]
  }