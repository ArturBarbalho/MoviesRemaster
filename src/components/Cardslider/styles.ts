import styled from 'styled-components'
import media from 'styled-media-query'
import {ArrowSmLeft} from '@styled-icons/heroicons-solid/ArrowSmLeft'
import {ArrowSmRight} from '@styled-icons/heroicons-solid/ArrowSmRight'
export const Wrapper = styled.section`
overflow-x:hidden;
max-width:1900px;
.slick-track,
    .slick-list {
      display: flex;
      margin-left:50px;
      margin-right:55px;
      ${media.lessThan('small')`
      margin:0;
      `}
    }
    .slick-prev,
    .slick-next {
      display: block;
      position: absolute;
      top: 50%;
      padding: 0;
      transform: translate(0, -50%);
    }
    
`
export const Left = styled(ArrowSmLeft)`
width: 60px;
color:lightgray;
cursor:pointer;
z-index:1;
`
export const Right = styled(ArrowSmRight)`
width: 60px;
color:lightgray;
cursor:pointer;
left:96vw;
${media.lessThan('large')`
left:94vw;
`}

`