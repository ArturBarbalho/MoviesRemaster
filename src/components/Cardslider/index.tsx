import { settings } from '../../sliderSettings/card'
import { CardProps } from '../../typesProps'
import Card from '../Card'
import Slider from '../Slider'
import * as S from './styles'

type props = {
  props:CardProps[]
}

const Cardslider = ({props}:props) => (
  <S.Wrapper >
    <Slider settings={settings} >
    {props.map((e)=>{
      return(
        <Card key={e.title} props={e} />
      )
    })}
    </Slider>
    
  </S.Wrapper>
)

export default Cardslider