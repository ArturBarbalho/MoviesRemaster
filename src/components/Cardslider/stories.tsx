import { Story, Meta } from '@storybook/react/types-6-0'
import Cardslider from '.'
import { CardMock } from '../../Mocks'
export default {
  title: 'Cardslider',
  component: Cardslider
} as Meta

const mock = [CardMock,CardMock,CardMock,CardMock,CardMock,CardMock,CardMock,CardMock]

export const Default: Story = () => <Cardslider props={mock} />