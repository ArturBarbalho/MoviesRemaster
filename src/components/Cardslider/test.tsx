import { render } from '@testing-library/react'
import '../../../.jest/match-media-mock'
import Cardslider from '.'
import { CardMock } from '../../Mocks'
const mock = [CardMock,CardMock,CardMock,CardMock,CardMock,CardMock,CardMock,CardMock,CardMock,CardMock]
describe('<Cardslider />', () => {
  it('should render the right number of slides', () => {
    const { container } = render(<Cardslider props={mock} />)

    expect(container.querySelectorAll('.slick-active')).toHaveLength(4)

    expect(container.firstChild).toMatchSnapshot()
  })
})