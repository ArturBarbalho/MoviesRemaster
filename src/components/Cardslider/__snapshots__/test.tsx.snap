// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cardslider /> should render the right number of slides 1`] = `
<section
  class="styles__Wrapper-sc-bipfd5-0 iLNsun"
  data-testid="warp"
>
  <section
    class="styles__Wrapper-sc-12nwagl-0 fTUMGo"
  >
    <div
      class="slick-slider slick-initialized"
      dir="ltr"
    >
      <span
        class="slick-arrow slick-prev"
        data-role="none"
        style="display: block;"
      >
         
        <svg
          aria-hidden="true"
          class="StyledIconBase-sc-ea9ulj-0 dXOYdB styles__Left-sc-bipfd5-1 dPtzgn"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9.707 14.707a1 1 0 0 1-1.414 0l-4-4a1 1 0 0 1 0-1.414l4-4a1 1 0 0 1 1.414 1.414L7.414 9H15a1 1 0 1 1 0 2H7.414l2.293 2.293a1 1 0 0 1 0 1.414z"
            fill-rule="evenodd"
          />
        </svg>
         
      </span>
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-10"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-9"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-8"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-7"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-6"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-5"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-4"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-3"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-2"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-1"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="1"
            style="outline: none;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="2"
            style="outline: none;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="3"
            style="outline: none;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="4"
            style="outline: none;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="5"
            style="outline: none;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="6"
            style="outline: none;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="7"
            style="outline: none;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="8"
            style="outline: none;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="9"
            style="outline: none;"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="10"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="11"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="12"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="13"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="14"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="15"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="16"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="17"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="18"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="19"
            tabindex="-1"
          >
            <div>
              <div
                class="styles__Wrapper-sc-1xbaftz-0"
              >
                <div
                  class="styles__Over-sc-1xbaftz-1 KCXaO"
                  data-testid="over"
                />
                <img
                  class="styles__Wrapper-sc-y6e3xk-0 hYSqaj"
                  src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        class="slick-arrow slick-next"
        data-role="none"
        style="display: block;"
      >
         
        <svg
          aria-hidden="true"
          class="StyledIconBase-sc-ea9ulj-0 dXOYdB styles__Right-sc-bipfd5-2 jiWySE"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.293 5.293a1 1 0 0 1 1.414 0l4 4a1 1 0 0 1 0 1.414l-4 4a1 1 0 0 1-1.414-1.414L12.586 11H5a1 1 0 1 1 0-2h7.586l-2.293-2.293a1 1 0 0 1 0-1.414z"
            fill-rule="evenodd"
          />
        </svg>
         
      </span>
    </div>
  </section>
</section>
`;
